universal-login-contracts/test/contracts/Proxy-refund.ts

```diff
@@ -0,0 +1,76 @@
+import chai, {expect} from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import {loadFixture, solidity, deployContract} from 'ethereum-waffle';
+import {transferMessage} from '../utils/ExampleMessages';
+import {utils, Contract, providers, Wallet} from 'ethers';
+import {calculateMessageSignature, UnsignedMessage, TEST_ACCOUNT_ADDRESS, ETHER_NATIVE_TOKEN, OPERATION_CALL, KeyPair} from '@universal-login/commons';
+import Loop from '../../build/Loop.json';
+import {encodeFunction} from '../utils';
+import {encodeDataForExecuteSigned} from '../../lib';
+import {walletContractWithFundsFixture} from '../fixtures/walletContract';
+
+chai.use(chaiAsPromised);
+chai.use(solidity);
+
+describe('CONTRACT: Proxy - refund', async  () => {
+  let provider: providers.Provider;
+  let signature: string;
+  let deployer: Wallet;
+  let mockToken: Contract;
+  let loopContract: Contract;
+  let infiniteCallMessage: UnsignedMessage;
+  let initialBalance: utils.BigNumber;
+  let keyPair: KeyPair;
+  let walletContract: Contract;
+
+
+  beforeEach(async () => {
+    ({provider, deployer, walletContract, keyPair, mockToken} = await loadFixture(walletContractWithFundsFixture));
+    loopContract = await deployContract(deployer, Loop);
+    const loopFunctionData = encodeFunction(Loop, 'loop');
+    infiniteCallMessage = {
+      from: walletContract.address,
+      to: loopContract.address,
+      value: utils.parseEther('0'),
+      data: loopFunctionData,
+      nonce: 0,
+      gasPrice: 1,
+      gasToken: '0x0',
+      gasLimit: utils.bigNumberify('240000'),
+      operationType: OPERATION_CALL
+    };
```

Comment:
Extract `createInfiniteCallMessage` (including `deployContract(deployer, Loop);`) and call it only test that actually use it.